### PR TITLE
[New Rule] Kubernetes Privileged Pod Created

### PR DESCRIPTION
**Pull request by imays11**
created toml for new rule and added to the non-ecs-schema with all fields

## Issues

https://github.com/elastic/detection-rules/issues/2031
## Summary
This rule detects when a user creates a new pod/container running in privileged mode. A highly privileged container has access to the node’s resources and breaks the isolation between containers. If compromised, an attacker can use the privileged container to gain access to the underlying host. Gaining access to the host may provide the adversary with the opportunity to achieve follow-on objectives, such as establishing persistence, moving laterally within the environment, or setting up a command and control channel on the host.
image

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/kubernetes/privilege_escalation_privileged_pod_created.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+]
+risk_score = 47
+rule_id = "c7908cac-337a-4f38-b50d-5eeb78bdb531"
+severity = "medium"
+tags = ["Elastic", "Kubernetes", "Continuous Monitoring", "Execution", "Privilege Escalation"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+kubernetes.audit.objectRef.resource : "pods" and kubernetes.audit.verb : "create" and kubernetes.audit.requestObject.spec.containers.securityContext.privileged : true
```
Comment (brokensound77) [inline rules/integrations/kubernetes/privilege_escalation_privileged_pod_created.toml:45]: nit

```suggestion
kubernetes.audit.objectRef.resource:pods and kubernetes.audit.verb:create and
  kubernetes.audit.requestObject.spec.containers.securityContext.privileged:true
```

**Review by terrancedejesus — APPROVED** [review]
LGTM. Outside of @brokensound77's comments, nothing else to add. Great stuff!

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/integrations/kubernetes/privilege_escalation_privileged_pod_created.toml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    access to all devices on the host. This allows the container nearly all the same access as processes running on the
+    host. An administrator may want to run a privileged container to use operating system administrative capabilities
+    such as manipulating the network stack or accessing hardware devices from within the cluster.
+    """,
+]
+index = ["logs-kubernetes.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Kubernetes Privileged Pod Created"
+note = """## Config
```
Comment (w0rk3r) [inline rules/integrations/kubernetes/privilege_escalation_privileged_pod_created.toml:30]:
```suggestion
note = """## Setup
```

**Pull request merged**